c10/util/Half-inl.h

```diff
@@ -48,9 +56,13 @@ inline C10_HOST_DEVICE Half::operator float() const {
   return __half2float(*reinterpret_cast<const __half*>(&x));
 #elif defined(__SYCL_DEVICE_ONLY__)
   return float(c10::bit_cast<sycl::half>(x));
+#else
+#if defined(CPU_CAPABILITY_AVX2) || defined(CPU_CAPABILITY_AVX512)
+  return at::vec::half2float_scalar(x);
```

Comment:
Should we just inline the code here?

Reply:
Alternatively we can unify code that works for all CPU types in the half2float, float2half conversion functions.

Reply:
> Should we just inline the code here?

I can not inline `at::vec::half2float_scalar` or `at::vec::float2half_scalar` in `c10/util/Half-inl.h` since PyTorch build systems does not add compilation flags related to AVX or AVX512 for `c10/util/Half-inl.h` during compilation. It is unable to compile AVX related instructions in `c10/util/Half-inl.h`. That's why I chose to add these two functions under aten/src/Aten/cpu.